lua/diagnostic.lua

```diff
@@ -90,7 +90,7 @@ function M.publish_diagnostics(bufnr)
 
   vim.schedule_wrap(function()
     vim.api.nvim_command("doautocmd User LspDiagnosticsChanged")
-  end)
+  end)()
```

Comment:
For a "fire-and-forget" single-use anonymous function like this, you can use `vim.schedule()` instead of `schedule_wrap`.

It would probably be better to either do that here or (better still) make the wrapped function once, store it, and then call it here, e.g.:

At global scope:
```lua
M.trigger_diagnostics_changed = vim.schedule_wrap(function()
  vim.api.nvim_command('doautocmd User LspDiagnosticsChanged')
end)
```

Here, in `publish_diagnostics`:
```lua
M.trigger_diagnostics_changed()
```